syntax = "proto2";
package common;

//学历
enum degree
{
    low         	= 1;	//专科以下
    major		 	= 2;	//专科
    college 		= 3;	//本科
    master			= 4;    //硕士
    doctor			= 5;    //博士及以上
}

//平台编号(appid)
enum platform_serial_number
{
    niiwoo = 10;    //你我金融
    tuandai = 11;   //团贷网
}

//性别
enum sex
{
    male   = 1; //男
    female = 2; //女
    unknow = 3; //不详
}

//  @deprecated 弃用, 新定义的接口请使用{ result_info }
enum err_code
{
    ASK_FAIL                                     = 0;             //成功
    ASK_SUCCEED                                  = 1;             //失败
}

//  @deprecated 弃用
message errorinfo
{
    optional  err_code error_code =1  [default = ASK_FAIL];
    optional  bytes error_info =2;
}

////////////////////////////////////////////////////////////////////////////
// 后续的接口中, 结果信息请使用 result_info, 原有的 error 字段后续需要更新
enum result_code_type
{
    FAILED                                   = -1;            // 失败
    SUCCESS                                  = 0;             // 成功
    TIMEOUT                                  = 1;             // 超时
    UNAUTHORIZED                             = 2;             // 未授权
}

message result_info
{
    optional result_code_type       result_code  = 1   [default = FAILED];
    optional bytes                  result_msg   = 2;
}
////////////////////////////////////////////////////////////////////////////


enum user_id_type
{
    other_id                                     = 0; // 其他
    identity_card_type                           = 1; // 身份证
    special_zone_permit                          = 2; // 港澳台特区通行证
    return_home_permit                           = 3; // 回乡通行证
    officers_card                                = 4; // 警官证
    extra_identity_card                          = 5; // 临时身份证
    residence_booklet                            = 6; // 户口簿
    soldiers_card                                = 7; // 士兵证
}

enum property_type
{
	house_property                               = 1; // 房产信息
	car_property                                 = 2; // 车产信息
	deposit_property                             = 3; // 储蓄资金
}

enum marriage_status
{
	unmarried                                    = 0; // 未婚
	married                                      = 1; // 已婚
	divorced                                     = 2; // 离异
	widowhood                                    = 3; // 丧偶
	other_status                                 = 4; // 未知
}

enum relation_type
{
	relative                                     = 1;  // 亲戚
	mate                                         = 2;  // 配偶
	children                                     = 3;  // 子女
	parent                                       = 4;  // 父母
	brothers_or_sisters                          = 5;  // 兄弟姐妹
	colleague                                    = 10; // 同事
	friend                                       = 11; // 朋友
	other_relation                               = 99; // 其他
}

enum tc_user_status
{
	non_authentication                           = 0;  // 未认证用户
	realname_certification                       = 1;  // 实名认证用户
	money_dabao                                  = 2;  // 钱大宝
	money_xiaobao                                = 4;  // 钱大宝
	borrower                                     = 8;  // 借款人
	partner                                      = 16;  // 合伙人
	in_work                                      = 32;  // 有业人士
	student                                      = 64;  // 学生用户
}

message identity_card
{
	optional user_id_type   id_type              = 1; // 证件类型
	optional bytes          id_card              = 2; // 证件号
	optional bytes          card_address         = 3; // 证件地址
}

message location_info
{
	optional bytes  province                     = 1; // 省份
	optional bytes  city                         = 2; // 城市
	optional bytes  area                         = 3; // 区域信息
	optional bytes  longitude                    = 4; // 精度
	optional bytes  latitude                     = 5; // 纬度
	optional bytes  event_time                   = 6; // 产生时间[yyyymmddHHMMSSmmmuuu]
	optional bytes  real_position                = 7; // 具体地址、理论上精确到具体街道等位置
}

// 学校信息
message school_info
{
	optional bytes 	collegeName                  = 1;  //学院
	optional degree eductionDegree               = 2;  //学历
	optional bytes 	educationCategory            = 3;  //学位
	optional bytes 	graduateYear                 = 4;  //毕业时间[yyyymmddHHMMSSmmmuuu]
	optional bytes 	enrollmentYear               = 5;  //入学时间[yyyymmddHHMMSSmmmuuu]
	optional bytes  major_subject                = 6;  //所学专业
	optional bytes  school_loc_privonce          = 7;  //学校所在省
	optional bytes  school_loc_city              = 8;  //学校所在城市
	optional bytes  school_loc_area              = 9;  //学校所在区域
	optional bytes  school_name                  = 10; //学校名称
	optional bytes  school_address               = 11; //学校详细地址

	optional bytes educonclusion                 = 12; // 学籍状态
	optional bytes eduform                       = 13; // 学习形式
	optional bytes edurecord                     = 14; // 层次
	optional bytes eduschool                     = 15; //学校
	optional bytes edutype                       = 16; //学历类型
	optional bytes source                        = 18; //来源
	optional bytes specialty                     = 19; //专业名称
}

// 关系人信息
message relation_contact
{
	optional relation_type  relationship         = 1; // 本人关系
	optional bytes          contact_name         = 2; // 关系人姓名
	optional bytes          mobilephone          = 3; // 电话号码
}

// 家庭信息
message family_info
{
	optional marriage_status  marriage           = 1; // 婚姻状态
	optional bool             childred_status    = 2; // 有无子女
	repeated relation_contact contact_list       = 3; // 关系人列表
}

// 资产信息
message property_info
{
	optional property_type  ptype                = 1; // 资产类型
	optional int32          valuation            = 2; // 资产估值
	optional bool           loan_status          = 3; // 资产是否有贷款
	optional int32          debt_info            = 4; // 资产的负债量借贷金额
	optional int32          age_limit            = 5; // 资产年限
	optional bytes          loc_province         = 6; // 资产省份位置
	optional bytes          loc_city             = 7; // 资产城市位置
	optional bytes          loc_area             = 8; // 资产区域位置
	optional int32          measurement          = 9; // 资产大小
	optional bytes          extend_info          = 10; // 扩展信息
}

// 工作信息
message work_info
{
	optional bytes          company_name         = 1;  // 公司名称
	optional bytes          position             = 2;  // 职位
	optional bytes          department           = 3;  // 部门信息
	optional location_info  company_loc          = 4;  // 公司位置
	optional bytes          industry             = 5;  // 工作行业(具体到行业名称)
	optional bytes          years                = 6;  // 工作年限
	optional bytes          income               = 7;  // 收入
	optional location_info  work_loc             = 8;  // 工作位置
}

message card_info
{
    optional bytes        bankcode              = 1;  // 银行名称
	optional bytes        cardno                = 2;  // 信用卡卡号
	optional bytes        accountname           = 3;  // 账户名
	optional bytes        email                 = 4;  // 邮箱
	optional double       maxLineofcredit       = 5;  // 最大信用额度
	optional double       maxreturnamount       = 6;  // 最大待还额度
	optional bytes        verifytime            = 7;  // 验证时间[yyyymmddHHMMSSmmmuuu]
	optional bytes        createtime            = 8;  // 添加时间[yyyymmddHHMMSSmmmuuu]
	optional bytes        updatetime            = 9;  // 修改时间[yyyymmddHHMMSSmmmuuu]
}

// 用户基本信息
message user_info
{
    optional bytes 	user_id                      = 1;  //用户ID
    optional identity_card id_card               = 2;  //用户证件
    optional bytes 	real_name                    = 3;  //账号名字
    optional bytes 	user_nickname                = 4;  //用户别名
    optional bytes 	mobile_phone                 = 5;  //用户手机号
    optional bytes 	email                        = 6;  //邮箱
    optional bytes 	address                      = 7;  //个人地址
    optional bytes 	bank_cart                    = 8;  //银行卡
    optional bytes 	ip                           = 9;  //手机ip
    optional bytes 	mac                          = 10; //
    optional bytes 	imei                         = 11; //设备号
    optional bytes 	blackBox                     = 12; //设备指纹
    optional bool  	is_prove                     = 13; //是否实名认证
    optional bytes  avatar                       = 14; //头像URL连接
    optional school_info  schoolinfo             = 15; //毕业学校信息
    optional family_info  family                 = 16; //家庭信息
    repeated property_info property              = 17; //资产信息

    optional bytes login_from_others             = 18; // 三方登陆来源
    optional bytes login_id                      = 19; // 三方登陆id
    optional bytes user_account                  = 20; // 用户登陆的账号
    optional tc_user_status user_status          = 21; // 用户身份
    optional common.sex  sex_type                = 22; // 性别
    optional bytes homtown                       = 23; // 籍贯
    optional bytes birthday                      = 24; // 出生年月日[yyyymmddHHMMSSmmmuuu]
    optional location_info living_loc_info       = 25; // 当前居住地位置信息
    optional int32 age                           = 26; // 用户年龄(按身份证的出生年月计算)
    optional work_info work_info_                = 27; // 工作信息

	optional int32 istuandai                     = 28; // 是否团贷网用户(1是，0否)',
	optional int32 isjcredit                     = 29; // 是否跑借款授信模型(1是，0否)
	optional int32 isdcredit                     = 30; // 是否跑担保授信模型(1是，0否)
	optional int32 isverify                      = 31; // 是否通过学历验证(1是，0否)
	optional int32 educationapplytype            = 32; // 学历提交类型(0未提交，1自动提交，2手动提交)
	optional bytes educationapplytime            = 33; // 学历提交时间(yyyymmddHHMMSSmmmuuu)
	repeated card_info  cardinfo                 = 34; // 信用卡列表
}

message user_basic_identification
{
    optional bytes identity_card                 = 1;   //用户身份证
    optional bytes mobile_phone                  = 2;   //手机号码
    optional bytes real_name                     = 3;   //身份证名字
}

message common_ask_header
{
    optional bytes response_topic                = 1;   // 目标回复主题
}
